url4aac7e6f55_simon0191_tetris3d_tgz-pJ8-Tetris3dJ8/src/vsdk/toolkit/io/image/ImagePersistence.java

```diff
@@ -0,0 +1,693 @@
+                fis.close();
+              }
+              catch ( Exception e ) {
+                  VSDK.reportMessage(null, VSDK.ERROR, "importRGBA",
+                                     "Cannot import image file \"" + inImageFd.getAbsolutePath() + "\"");
+                return null;
+            }
+            AwtRGBAImageRenderer.importFromAwtBufferedImage(bi, retImage);
+
+            return retImage;
+        }
+        throw new ImageNotRecognizedException("Image not recognized", inImageFd);
+    }
+
+    /**
+    Given a text string, this method determines if is a Unix style single
+    line comment, that is, if line begins with a '#' character.
+    */
+    private static boolean
+    isTextComment(String line)
+    {
+        char arr[] = line.toCharArray();
+        int i;
+
+        for ( i = 0; i < arr.length && (arr[i] != ' ' && arr[i] != '\t'); i++ );
+
+        if ( i < arr.length && arr[i] == '#' ) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+    Given the filename of an input data file which contains an image, this
+    method tries to recognize the file format and load the contents of it
+    to the image.
+
+    @todo Do not assume the file format only from the filename extension,
+    but trying to detect file headers.
+
+    @param imagen - The file respesenting the image
+    @return An RGBImage entity that contains the image loaded in memory.
+
+    Will change:
+      - Choose a better name for this method
+      - Do not recieve a File, but a Stream of bytes
+    */
+    public static RGBImage importRGB(File inImageFd) throws ImageNotRecognizedException, Exception
+    {
+        String type = extractExtensionFromFile(inImageFd);
+        RGBImage retImage = new RGBImage();
+
+        //- Try optimized reading, if native library is available ---------
+        if ( type.equals("png") ) {
+            nativeWarningGiven = true;
+        }
+
+        //-----------------------------------------------------------------
+        if( type.equals("tga") ) {
+            TargaImage t = new TargaImage(inImageFd);
+            t.exportRGB(retImage);
+            return retImage;
+        }
+        else if( type.equals("jpg") || type.equals("jpeg") ||
+                 type.equals("gif") || type.equals("png") )  {
+            BufferedImage bi = null;
+
+            // OLD SLOW METHOD, DO NOT USE!
+            //java.awt.Toolkit awtTools = java.awt.Toolkit.getDefaultToolkit();
+            //java.awt.Image image;
+            //image = awtTools.getImage(inImageFd.getAbsolutePath());
+            //bi = toBufferedImage(image);
+
+            try {
+                BufferedInputStream bis;
+                FileInputStream fis;
+
+                fis = new FileInputStream(inImageFd);
+                bis = new BufferedInputStream(fis);
+                bi = ImageIO.read(bis);
+                bis.close();
+                fis.close();
+              }
+              catch ( Exception e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 271.